core/src/main/java/hudson/model/Queue.java

```diff
@@ -799,8 +799,9 @@ private WaitingItem peek() {
     }
 
     private List<Item> filterItemListBasedOnPermissions(List<Item> r, Item t) {
-        if (t.task instanceof hudson.model.Item) {
-            if (((hudson.model.Item)t.task).hasPermission(hudson.model.Item.READ)) {
+        if (t.task instanceof hudson.security.AccessControlled) {
```

Comment:
Would be nice to adjust `Task` Javadoc to note that it is encouraged to implement `AccessControlled`, and noting that `checkAbortPermission` and `hasAbortPermission` should just check `Item.CANCEL` permission.